week1/homework/src/server.js

```diff
@@ -2,14 +2,47 @@
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
+/* `createServer` MUST return an instance of `http.Server` 
+otherwise the tests will fail.
  */
-function createServer(port) {
+function createServer() {
   let state = 10;
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        const stateJson1 = { 'state': state };
+        response.writeHead(200, { 'Content-Type': 'application/json' });
+        response.end(JSON.stringify(stateJson1));
+        break;
+
+      case '/add':
+        state++;
+        const stateJson2 = { 'state': state };
+        response.writeHead(200, { 'Content-Type': 'application/json' });
```

Comment:
You are adding a header to every `case` and writing the state as JSON to the response. You could put this in a function, witch you can call from the different `case`s. 

In that way, if you would like to change the key for 'state' in `{'state': state}` into `{'the_state': state}` for example, you have to only do it in one place.